### PR TITLE
Extend initial project structure and related resources

### PR DESCRIPTION
**Pull request by tpluscode**
This largish PR builds the foundation for project metadata:

1. `cc:namespace` which will serve for cube URI etc
2. generate the URI for theProject's `cc:cubeGraph` property
3. create a `void:Dataset` resource to hold metadata and cubes
4. CSV mapping project creates the initial cube
5. Bootstrap cube metadata resource

### CODE REVIEW

File: apis/core/test/domain/table/create.test.ts [hunk 1]
```diff
@@ -12,7 +12,9 @@ describe('domain/table/create', () => {
   const tableCollection = clownface({ dataset: $rdf.dataset(), term: $rdf.namedNode('tables') }).addOut(cc.csvMapping, $rdf.namedNode('myCsvMapping'))
 
   beforeEach(() => {
-    store = new TestResourceStore([])
+    store = new TestResourceStore([
+      tableCollection,
+    ])
   })
 
   it('creates identifier by slugifying schema:name', async () => {
```
Comment (lucafurrer) [inline apis/core/test/domain/table/create.test.ts:20]: Unit test fails, do you need to call the persist of the created resource first?

**Review by lucafurrer — CHANGES_REQUESTED** [review]
Looks fine to me, except the failing unit test.

File: apis/core/bootstrap/shapes/cube-project.ts [hunk 2]
```diff
@@ -26,7 +26,14 @@ ${shape('cube-project/create')} {
         ${shape('cube-project/create#ExistingCube')}
       ) ;
       ${sh.defaultValue} ${shape('cube-project/create#CSV')} ;
-      ${sh.order} 1 ;
+      ${sh.order} 20 ;
+    ] ;
+    ${sh.property} [
+      ${sh.name} "Cube namespace" ;
+      ${sh.path} ${cc.namespace} ;
```
Comment (martinmaillard) [inline apis/core/bootstrap/shapes/cube-project.ts:33]: We'll need a `sh:description` for this one

File: apis/core/lib/domain/cube-projects/create.ts [hunk 3]
```diff
@@ -18,57 +20,28 @@ export async function createProject({
...
-    .createMember(projectsCollection, id.cubeProject(label.value))
-
-  project.addOut(rdfs.label, label)
-    .addOut(dcterms.creator, user)
-
-  if (shape('cube-project/create#CSV').equals(resource.out(cc.projectSourceKind).term)) {
-    const mapping = store
-      .create(id.csvMapping(project))
-      .addOut(rdf.type, cc.CsvMapping)
+}: CreateProjectCommand): Promise<Project.Project> {
```
Comment (martinmaillard) [inline apis/core/lib/domain/cube-projects/create.ts:23]: Now the code to create a new project is split between the handler, this function and the mixin classes. Is there a reasoning to know what should go where?

**Review by martinmaillard — APPROVED** [review]

**Pull request merged**